### ISSUE TITLE
avoid-leaking-state-in-ember-objects flags computed.sort definition

### ISSUE THREAD

**Issue by bryanhickerson** [ember-cli/eslint-plugin-ember#192]
If you use the sort computed property, it gets flagged as leaking state. Example:
```
sortedTimeslotBlocks: computed.sort('timeslotBlocks', 'timeslotBlocksSorting'),
timeslotBlocksSorting: ['startsAt:asc'],
```
I can see why this was flagged, but it seems weird to flag this since the docs use this syntax: https://emberjs.com/api/ember/2.15/namespaces/Ember.computed/methods/sort?anchor=sort

**Comment by backspace** [comment]
yeah, this is confusing to me since it’s in the official documentation, and the way to address this seems pretty weird in some situations. For instance, in [this Mirage serialiser](https://github.com/backspace/prison-rideshare-ui/blob/2d9d24f2c61f9b70329f2741049781f5efa64f15/mirage/serializers/person.js#L4), is it really a danger?

**Comment by xtagon** [comment]
As to whether it's really a danger, it probably depends on whether you mean "technically" or "in practice" -- if none of your code ever modifies the sorting array property, then leakage shouldn't be a concern. If your code does modify it, then you just have to be aware of the leaking effect.

**Comment by st-h** [comment]
same seems to be the case for configuring serializers using the attrs property as explained here https://guides.emberjs.com/v2.18.0/models/customizing-serializers/ I have also seen warnings regarding queryParams and other binding related stuff.
I am afraid this check will need quite a number of exceptions or we need to deal with using ember fundamentally different than what is described in the docs today to make this work reliably.

**Comment by rwjblue** [comment]
We should update the guides and documentation as we find it, to ensure that it does **not** get flagged by the rule. I submitted https://github.com/emberjs/ember.js/pull/16084 to address the link in the original issue description.

**Comment by EvanHammond** [comment]
This is also the case with the docs for [Components](https://guides.emberjs.com/v2.18.0/components/customizing-a-components-element/) (see: `classNames`, `classNameBindings`, and `attributeBindings`).

**Comment by xtagon** [comment]
There are a lot of different ways to solve it with different syntax. Is there merit to one over the other? For example, is `Object.freeze` recommended over using constants or init?

**Comment by rwjblue** [comment]
Yes, however `classNames`, `classNameBindings`, and `attributeBindings` are very common and as such they are whitelisted in this rule ([see here](https://github.com/ember-cli/eslint-plugin-ember/blob/master/lib/rules/avoid-leaking-state-in-ember-objects.js#L6-L16)).
